WordPress/gutenberg
# Components: update `<Card />` implementation to g2 version

## Description


- Replace the current implementation of the `<Card />` component with the new g2 implementation (which, prior to this PR, sits in [`packages/components/src/ui/card`](https://github.com/WordPress/gutenberg/tree/4857ad58c1241b3d63d21a6880c989b85746c3dc/packages/components/src/card).
- Keep all components and their existing APIs unchanged. The only exceptions are:
  -  The `isElevated` prop on the `<Card />` component is being soft-deprecated in favour of a new `elevation` prop (which enables fine-grain control on the card's elevation)
  - The `<CardMedia />` component didn't exist in the g2 implementation and was created directly in this PR (pretty much a 1:1 with the current version)
  - The `<CardDivider />` component didn't exist in the g2 implementation and was created directly in this PR. Although its APIs are the same as the old version, internally it now extends the `<Divider />` component
  - The border color has slightly changed to match the same color used by other g2 components
  - There's a new possible value for the `size` property: `none`
  - The CSS `padding` associated to the `size` property has slightly changed to match the spacing system used by other g2 components (used by `<Card />`, `<CardBody />`, `<CardHeader />` and `<CardFooter />`)

<details>
<summary>Detailed action plan (used to track the PR's progress)</summary>

1. ~Replace the old implementation with the new from `ui/` folder~ ✅
2. ~Make project compile (fix imports, etc)~ ✅
3. ~Reconciliate APIs:~ ✅
    1. ~Match new implementation with OLD APIs~ ✅
        - ~Add and deprecate `isElevated`~ ✅
        - ~Add `size` and propagate it to the sub components~ ✅
        - ~Propagate `isBorderless` to sub components~ ✅
        - ~Remove the `size` property on CardHeader~ ✅
        - ~Remove the `CardInnerBody` component~ ✅
    2. ~Refactor folder structure like Flex (one folder per component)~ ✅
    3. ~Add missing subcomponents~ ✅
        - ~`CardMedia`~ ✅
        - ~`CardDivider`~ ✅
    4. ~Replace Storybook story with a single story which includes `<Card />` and all of its subcomponents in one place~ ✅
    5. ~Update tests~ ✅
    6. ~Update READMEs~ ✅
4. ~Update the CHANGELOG:~ ✅
    - ~Internal: "rewrote component's internals"~ ✅
    - ~External: `isElevated` deprecation, added `none` as a value for `size`,  changed border color for header/footer/divider~ ✅
5. ~Search and update any component's usages across the codebase~ ✅
6. ~Remove dependencies from the`/ui/` folder~ ✅
    - ~Promote `<Surface />` component #32439~ ✅
    - ~Promote `<Scrollable />` component #32446~ ✅
    - ~Make sure there's no other references to `ui/` apart from context~ ✅
</details>

## Testing instructions




- [ ] Project compiles, all tests pass
- [ ] Compare storybook between this branch and production
    - [Current production `Card` Storybook](https://wordpress.github.io/gutenberg/?path=/story/components-card--default)
    - New `Card` Storybook:
        - Pull this PR branch on your machine and run `npm run storybook:dev`
        - Visit [`http://localhost:50240/?path=/story/components-card--default`](http://localhost:50240/?path=/story/components-card--default)
- [ ] Check old vs new READMEs and make sure that the APIs make sense
- [ ] Make sure that usages of `<Card />` in the repo keep working as before

## Screenshots 

<img width="376" alt="Screenshot 2021-06-02 at 18 28 01" src="https://user-images.githubusercontent.com/1083581/120517026-4731ec00-c3d0-11eb-9938-97d79eb61505.png">

## Types of changes





New feature

## Checklist:
